Documentation/Signoffs/Microcontroller.md

```diff
@@ -0,0 +1,51 @@
+
+These jobs require a sufficiently powerful board.  One that has ample storage, plenty of physical space, decent processing power, sufficient support, and none restrictive.  Before choosing components and detailing the design, the specific constraints of the subsystem need to be known and laid out.  Then any potential solutions can be compared to the constraints to test viability and fulfillment of the subsystem’s purpose. 
+
+## CONSTRAINTS ON THE SUBSYSTEM
+The constraints given below are generalized to allow for comparison of any potential contender for the microcontroller.  Constraints on the subsystem are as follows:
+1. With a budget of $300, a super-powered board is not affordable.  The board price ranges between $20-$60.  This range is gathered from estimating the costs of other components and viewing prices of avaiable options that meet the needs of the subsystem.
+2. The microcontroller must support communication between components.  To do this, Serial Peripheral Interface (SPI), Universal Asynchronous Receiver/Transmitter (UART), or some form of hardware communication will need to be achievable on the board.  SPI is a synchronous method while UART is an asynchronous method for two devices connected over a short distance to liaise.
+3. External storage support is required of this subsystem. The most preferred form of external storage is a Universal Serial Bus (USB) Flash Drive as it is common in households and easily available.
+4. A major goal of the project is to allow for synchronous operations of at least 3 air quality sensors.  Sensors may be either analog or digital, so there must be support for both.  The board must have enough Input/Output (I/O) pins to operate any combination of 3 sensors whether they be digital or analog.   Different sensors interface with a potentially different number of pins.  Preliminary research shows that a sensor may take up to 5 pins to operate.  Other sensors may have more, but it is more common for 5 and under.  This means the board must have at least 15 digital input pins and 15 analog input pins.
+5. The cpu clock speed does not need to be very fast.  The code will be mostly Input/Output heavy with less computation.  This means that there will be a lot of waiting for responses and does not require a high-speed processor.  A cpu clock speed of at least 1 MHz is preferred for fast bootup.  Cpu's over 1 MHz are very common and just provides a reliable speed for what computation is present.
+6. There must be sufficient memory for the code.  It is unknown how much memory the code will occupy.  Comparing similar operating code with libraries and compilation show slightly over or under 150 KB (Kilobytes) in Flash/Disk/Storage.  To account for the worst case, the board will need over 150 KB.
```

Comment:
If there must be sufficient memory but the amount of memory needed is unknown, then appropriate efforts need to be made to estimate the amount of needed memory.

Reply:
Dr. Austen agreed.  We have made an estimating range based on similarly sized code and compilation of related libraries.